jet.py

```diff
@@ -53,54 +53,96 @@ class JetCorrProducer:
-    def __init__(self, period,isData):
-        JEC_SF_path_period = JetCorrProducer.JEC_SF_path.format(period)
-        JEC_dir = directories_JEC[period]
-        JEC_SF_db = "Corrections/data/JECDatabase/textFiles/"
-
-        JER_dir = directories_JER[period]
-        JER_SF_db = "Corrections/data/JRDatabase/textFiles/"
-
-        JER_SF_txtPath_MC = f"{JER_SF_db}/{JER_dir}_MC/{JER_dir}_MC_SF_AK4PFchs.txt"
-        JER_PtRes_txtPath_MC = f"{JER_SF_db}/{JER_dir}_MC/{JER_dir}_MC_PtResolution_AK4PFchs.txt"
-        JER_PhiRes_txtPath_MC = f"{JER_SF_db}/{JER_dir}_MC/{JER_dir}_MC_PhiResolution_AK4PFchs.txt"
-        JER_EtaRes_txtPath_MC = f"{JER_SF_db}/{JER_dir}_MC/{JER_dir}_MC_EtaResolution_AK4PFchs.txt"
-
-        JER_SF_txtPath_data = f"{JER_SF_db}/{JER_dir}_DATA/{JER_dir}_DATA_SF_AK4PFchs.txt"
-        JER_PtRes_txtPath_data = f"{JER_SF_db}/{JER_dir}_DATA/{JER_dir}_DATA_PtResolution_AK4PFchs.txt"
-        JER_PhiRes_txtPath_data = f"{JER_SF_db}/{JER_dir}_DATA/{JER_dir}_DATA_PhiResolution_AK4PFchs.txt"
-        JER_EtaRes_txtPath_data = f"{JER_SF_db}/{JER_dir}_DATA/{JER_dir}_DATA_EtaResolution_AK4PFchs.txt"
-
-        JEC_Regouped_txtPath_MC = f"{JEC_SF_db}/{JEC_dir}/{regrouped_files_names[period]}"
-
-        JetCorrProducer.isData = isData
-
-        ptResolution = os.path.join(os.environ['ANALYSIS_PATH'],JER_PtRes_txtPath_MC.format(period))
-        ptResolutionSF = os.path.join(os.environ['ANALYSIS_PATH'],JER_SF_txtPath_MC.format(period))
-        JEC_Regrouped = os.path.join(os.environ['ANALYSIS_PATH'], JEC_Regouped_txtPath_MC.format(period))
-        if JetCorrProducer.isData:
-            ptResolution = os.path.join(os.environ['ANALYSIS_PATH'],JER_PtRes_txtPath_data.format(period))
-            ptResolutionSF = os.path.join(os.environ['ANALYSIS_PATH'],JER_SF_txtPath_data.format(period))
-        if not JetCorrProducer.initialized:
-            ROOT.gSystem.Load("libJetMETCorrectionsModules.so")
-            ROOT.gSystem.Load("libCondFormatsJetMETObjects.so")
-            ROOT.gSystem.Load("libCommonToolsUtils.so")
-            headers_dir = os.path.dirname(os.path.abspath(__file__))
-            header_path = os.path.join(headers_dir, "jet.h")
-            JME_calc_base = os.path.join(headers_dir, "JMECalculatorBase.cc")
-            JME_calc_path = os.path.join(headers_dir, "JMESystematicsCalculators.cc")
-            ROOT.gInterpreter.Declare(f'#include "{JME_calc_base}"')
-            ROOT.gInterpreter.Declare(f'#include "{JME_calc_path}"')
-            ROOT.gInterpreter.Declare(f'#include "{header_path}"')
-            ROOT.gInterpreter.ProcessLine(f"""::correction::JetCorrProvider::Initialize("{ptResolution}", "{ptResolutionSF}","{JEC_Regrouped}", "{periods[period]}")""")
-            JetCorrProducer.period = period
-            JetCorrProducer.initialized = True
-
-    def getP4Variations(self, df, source_dict, apply_JER=True, apply_JES=True):
+    def __init__(self, period, isData):
+        self.year = int(period[:4])
```

Comment:
Don't rely on `period` - in the future, Run 2 should also move to corrib. Pass `use_corrlib`argument to `__init__` instead. You should also provide an argument what JEC unc grouping should be used. E.g. `OneJES` (meaning total only), `FullJES` (using all 27 sources), and `CombinedJES` (using combined unc set).